projects/packages/search/tests/php/test-rest-controller.php

```diff
@@ -348,4 +349,105 @@ public function test_deactivate_plan_editor() {
 		$this->assertEquals( 403, $response->get_status() );
 	}
 
+	/**
+	 * Testing the `GET /jetpack/v4/search/local-stats` endpoint with blog_token.
+	 */
+	public function test_get_post_type_breakdown() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/search/local-stats' );
+		$request->set_header( 'content-type', 'application/json' );
+
+		$response = $this->dispatch_request_signed_with_blog_token( $request );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals(
+			array(
+				'post_count'          => 0,
+				'post_type_breakdown' => array(),
+			),
+			$response->get_data()
+		);
+	}
+
+	/**
+	 * Testing the `GET /jetpack/v4/search/local-stats` with editor user.
+	 */
+	public function test_get_post_type_breakdown_regular_user() {
+		wp_set_current_user( $this->editor_id );
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/search/local-stats' );
+		$request->set_header( 'content-type', 'application/json' );
+
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 403, $response->get_status() );
+	}
+
+	/**
+	 * Testing the `GET /jetpack/v4/search/local-stats` with no auth.
+	 */
+	public function test_get_post_type_breakdown_no_auth() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/search/local-stats' );
+		$request->set_header( 'content-type', 'application/json' );
+
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	/**
+	 * Signs a request with a blog token before dispatching it.
+	 *
+	 * Ensures that these tests pass through Connection_Rest_Authentication::wp_rest_authenticate,
+	 * because WP_REST_Server::dispatch doesn't call any auth logic (in a real
+	 * request, this would all happen earlier).
+	 *
+	 * @param WP_REST_Request $request The request to sign before dispatching.
+	 * @return WP_REST_Response
+	 */
+	private function dispatch_request_signed_with_blog_token( $request ) {
+		add_filter( 'jetpack_options', array( $this, 'mock_jetpack_site_connection_options' ), 10, 2 );
+
+		$token     = 'new:1:0';
+		$timestamp = (string) time();
+		$nonce     = 'testing123';
+		$body_hash = '';
+
+		$_SERVER['REQUEST_METHOD'] = 'POST';
+
+		$_GET['_for']      = 'jetpack';
+		$_GET['token']     = $token;
+		$_GET['timestamp'] = $timestamp;
+		$_GET['nonce']     = $nonce;
+		$_GET['body-hash'] = $body_hash;
+		// This is intentionally using base64_encode().
+		// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+		$_GET['signature'] = base64_encode(
+			hash_hmac(
+				'sha1',
+				implode(
+					"\n",
+					array(
+						$token,
+						$timestamp,
+						$nonce,
+						$body_hash,
+						'POST',
+						'anything.example',
+						'80',
+						'',
+					)
+				) . "\n",
+				'blogtoken',
+				true
+			)
+		);
```

Comment:
This test setup looks brittle. Will this break if our blog token signing approach changes in the future?

Reply:
Good question. I don't think the approach is easily going to change in the future unless the protocol has defects etc, because the packages are all out there in the wild and we don't want them to break. Besides this is widely used in several other place including in the connection, backup, and social packages. Lastly it's just tests, we definitely could afford the breakage here, if it's changed, that's what it's for. 😃